src/hdmf_zarr/backend.py

```diff
@@ -837,7 +1009,7 @@ def write_dataset(self, **kwargs):  # noqa: C901
         # Exhaust the DataChunkIterator if the dataset was given this way. Note this is a no-op
         # if the self.__dci_queue is empty
         if exhaust_dci:
-            self.__dci_queue.exhaust_queue()
+            self.__dci_queue.exhaust_queue(number_of_jobs=number_of_jobs)
```

Comment:
Should this use the 'max_threads_per_process' and 'multiprocessing_context' kwargs?

Reply:
I adjusted things towards @oruebel's recommendation on attributes of an instance of the Queue class

I wasn't quite sure if it would work so I tried it on a separate branch #118, though it appears to be working, so switching to that

It does end up simplifying a lot of these docvals and argument passing

Reply:
> it appears to be working, so switching to that
> 
> It does end up simplifying a lot of these docvals and argument passing

Thanks for all your efforts on this and your patients with all the changes and discussions. I'm glad that our discussions are helping to make the PR better, but I do appreciate that this has taken a lot of effort on your end. But I think this is coming along nicely and that this should be ready to merge soon.